A rule that rejects obvious words in the commit title

tehraninasab opened the pull request:
Replaced https://github.com/nblockchain/conventions/pull/15

tehraninasab committed:
commitlint/plugins.test: add failing tests

Add tests for reject-obvious-words rule.

knocte commented:
Let's consider the commitlint PR for fixing subject-full-stop highest priority, but after you've tackled that, you can address the feedback that I'm going to share about this PR.

knocte commented on commitlint/plugins.test.ts:
```diff
@@ -652,6 +652,30 @@ test("proper-issue-refs3", () => {
+        commitMsgWithoutObviousWordAfterColon
+    );
+    expect(rejectObviousWord3.status).toBe(0);
+});
```
@realmarv I've slightly changed my mind of how this rule should work, but instead of explaining it, I'll just request you to add the following tests:
- `foo: update bla bla` -> status should not be 0
- `foo: update` -> status should be 0
- `foo: change` -> status should not be 0
- `foo: update bla bla\n\nBla blah.` -> status should be 0
- `foo: change bla bla\n\nBla blah.` -> status should not be 0

That is, there is an exception to the rule: the word update can go alone after area ("change" or "modify" cannot). The word `update` can only go with extra words after it if body length is higher than zero ("change" or "modify" cannot).

The above means that you should not make this rule extensible (i.e. by putting the list of words in a separate file or list). You have to hardcode them in the plugin instead.

realmarv replied:
Done

tehraninasab committed:
commitlint: add new rule

Add a rule to reject obvious words in the commit title.

knocte commented:
Please add a new failing test with this commit message: https://github.com/realmarv/conventions/commit/006fe222f6911bb3a522325388f6ee339e986120

That is, if the subject contains just the word "fix", and there's no commit msg body, this rule should complain as well.

tehraninasab committed:
commitlint/plugins.test: add failing test

tehraninasab committed:
commitlint/plugins: fix the rule

If the subject contains just the word "fix", and there's no
commit msg body, this rule should complain.

realmarv commented:
> Please add a new failing test with this commit message: [realmarv@006fe22](https://github.com/realmarv/conventions/commit/006fe222f6911bb3a522325388f6ee339e986120)
> 
> That is, if the subject contains just the word "fix", and there's no commit msg body, this rule should complain as well.

Done
